### PR TITLE
fixes OSError: [Errno 22] Invalid argument

### LINKED ISSUE #2220: Potential OSError: [Errno 22] Invalid argument
In current `proc.py:CommandPipeline` implementation, we built and run procs first, then give controlling terminal to them. There are cases that all the processes have finished, then the "give terminal to them" will raise errors.

```
$ ls
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/xonsh/base_shell.py", line 322, in default
    run_compiled_code(code, self.ctx, None, 'single')
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/xonsh/codecache.py", line 65, in run_compiled_code
    func(code, glb, loc)
  File "<xonsh-code>", line 1, in <module>
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/xonsh/built_ins.py", line 858, in subproc_captured_hiddenobject
    return run_subproc(cmds, captured='hiddenobject')
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/xonsh/built_ins.py", line 799, in run_subproc
    command = HiddenCommandPipeline(specs)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/xonsh/proc.py", line 1756, in __init__
    if update_fg_process_group(pipeline_group, background):
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/xonsh/proc.py", line 1691, in update_fg_process_group
    return give_terminal_to(pipeline_group)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/xonsh/jobs.py", line 138, in give_terminal_to
    os.tcsetpgrp(FD_STDERR, pgid)
OSError: [Errno 22] Invalid argument
```

### PR DESCRIPTION
**Pull request by mitnk**
closes #2220

### CODE REVIEW

File: xonsh/jobs.py [hunk 1]
```diff
@@ -135,9 +135,19 @@ def give_terminal_to(pgid):
...
             oldmask = signal.pthread_sigmask(signal.SIG_BLOCK,
                                              _block_when_giving)
-            os.tcsetpgrp(FD_STDERR, pgid)
-            signal.pthread_sigmask(signal.SIG_SETMASK, oldmask)
-            return True
+            try:
+                os.tcsetpgrp(FD_STDERR, pgid)
+                return True
+            except OSError as e:
+                if getattr(e, 'errno') == 22:  # [Errno 22] Invalid argument
```
Comment (scopatz) [inline xonsh/jobs.py]: I think the getattr() call should either have a default value or it should be `e.errno`.  The current method doesn't add any safety and OSErrors should always have the `errno` attribute.

**Comment by scopatz** [comment]
Thanks for putting this in @mitnk!

File: xonsh/jobs.py (hunk 1 above)
Comment (mitnk) [inline xonsh/jobs.py]: Ah, right! :)

**Comment by scopatz** [comment]
Thanks @mitnk!

**Pull request merged**